book/src/rfc/const_generic.md

```diff
@@ -0,0 +1,265 @@
+## Summary
+Enable const generic to improve modularization and reusability of the code.
+
+## Motivation
+In the current implementation, array initialization requires explicitly stating all element values, which becomes impractical for large arrays. Furthermore, the lack of support for constant variables as array sizes limits the ability to write functions that generically handle arrays of varying sizes. This also necessitates hardcoding loop ranges, reducing flexibility.
```

Comment:
not-really-important comment: it's always a good idea to add example. For example, here you could add an example of what array initialization you're talking about. I think also in general it'd be nice to have the RFC be its own PR so that people can comment/improve on it without messing with the implementation PR. But let's keep this way this time :)